tempesta_fw/connection.c

```diff
@@ -109,12 +109,9 @@ tfw_connection_new(struct sock *sk, int type, void *handler,
 	}
 
 	sk->sk_user_data = conn;
-
-	conn->sk_destruct = sk->sk_destruct;
```

Comment:
Because re-placing `sk_destruct()` with a custom handler and then calling the original `sk_destruct()` function was a dirty solution. The `sk->sk_destruct` callback is usually occupied with a specific destructor of linux TCP/IP stack, like `inet_release()`. What we did by hooking up the callback is kinda monkey patching, the `sk_destruct` is intended for use with lower level code, and shall not be touched by us.

Now, instead of re-placing the `sk_destruct` and calling it manually later, we have our custom callback in `TfwConnection` which is called from `tfw_connection_close()`. Basically it is cleaner because we don't have to call original `sk_destruct` now, and it doesn't violate encapsulation of lower-level code.

Actually, this little fix is not required for this pull-request. Originally it was, because the callback returned a value that allowed to not deallocate the socket, but then I changed the implementation and the return value was not required anymore, but the callback is survived. I left it because it is a little bit cleaner than calling `sk_destruct` as I described above.